packages/expo-cli/src/commands/init.js

```diff
@@ -164,52 +178,108 @@ async function shouldUseYarnAsync() {
-      name: 'useYarn',
-      message: `Yarn v${version} found. Use Yarn to install dependencies?`,
-    });
+    let answer = await prompt(
+      {
+        type: 'confirm',
+        name: 'useYarn',
+        message: `Yarn v${version} found. Use Yarn to install dependencies?`,
+      },
+      {
+        nonInteractiveHelp:
+          'Please specify either --npm or --yarn to choose the installation method.',
+      }
+    );
     return answer.useYarn;
   } catch (e) {
     return false;
   }
 }
 
 async function promptForWorkflowAsync() {
-  let answer = await prompt({
-    type: 'list',
-    name: 'workflow',
-    message: 'Choose which workflow to use:',
-    choices: [
-      {
-        value: 'managed',
-        name:
-          chalk.bold('managed') +
-          ' (default)' +
-          '\n' +
-          wordwrap(2, process.stdout.columns || 80)(
-            'Build your app with JavaScript with Expo APIs.'
-          ),
-        short: 'managed',
-      },
+  let answer = await prompt(
+    {
+      type: 'list',
+      name: 'workflow',
+      message: 'Choose which workflow to use:',
+      choices: [
+        {
+          value: 'managed',
+          name:
+            chalk.bold('managed') +
+            ' (default)' +
+            '\n' +
+            wordwrap(2, process.stdout.columns || 80)(
+              'Build your app with JavaScript with Expo APIs.'
+            ),
+          short: 'managed',
+        },
 
-      {
-        value: 'advanced',
-        name:
-          chalk.bold('advanced') +
-          ' (experimental 🚧)' +
-          '\n' +
-          wordwrap(2, process.stdout.columns || 80)(
-            'Build your app with JavaScript with Expo APIs and custom native modules.'
-          ),
-        short: 'advanced',
-      },
-    ],
-  });
+        {
+          value: 'advanced',
+          name:
+            chalk.bold('advanced') +
+            ' (experimental 🚧)' +
+            '\n' +
+            wordwrap(2, process.stdout.columns || 80)(
+              'Build your app with JavaScript with Expo APIs and custom native modules.'
+            ),
+          short: 'advanced',
+        },
+      ],
+    },
+    {
+      nonInteractiveHelp:
+        '--workflow: argument is required in non-interactive mode. Valid choices are: managed, advanced.',
+    }
+  );
   return answer.workflow;
 }
 
-async function promptForInitialConfig(parentDir, dirName, workflow) {
+async function promptForInitialConfig(parentDir, dirName, workflow, options) {
+  if (options.parent && options.parent.nonInteractive) {
+    let config = {
+      slug: dirName,
+    };
+    if (!isString(options.name) || options.name === '') {
+      throw new CommandError(
+        'NON_INTERACTIVE',
+        '--name: argument is required in non-interactive mode.'
+      );
+    } else {
+      config.name = options.name;
+    }
+
+    if (workflow === 'advanced') {
+      if (!isString(options.androidPackage) || options.androidPackage === '') {
```

Comment:
`options` gathered by commander are always strings unless you explicitly specify a function to convert the value - http://tj.github.io/commander.js/#Command.prototype.option
It doesn't make sense to do `!isString(options.androidPackage) || options.androidPackage === ''` and `!options.androidPackage` is the same check.

Reply:
This is a workaround for a quirk of Commander.js that I run into: if the user runs `expo init hello --non-interactive --android-package` (without a value given to `--android-package`), Commander assumes `--android-package` is a boolean flag and `options.androidPackage` gets set to `true`.

Reply:
Fair enough.